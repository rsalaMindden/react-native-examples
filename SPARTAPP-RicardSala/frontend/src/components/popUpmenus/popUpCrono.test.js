import React from "react";
import Training from './popUpCrono'
import {render,fireEvent}from '../../utils/tetsUtils/test.utils'

const navigation={
    pop:jest.fn(),
    navigate:jest.fn()
}


describe('Given a Training component',()=>{
    describe('When it is render',()=>{
        let screen
        beforeEach(()=>{
            screen=render(<Training navigation={navigation}/>)
        })
        test('Then should match the snapshot',()=>{
            expect(screen).toMatchSnapshot();
          })
  
          describe('And FOR TIME button is clicked', () => {
            test('Then navigation.navigate should have been called with handleClick', () => {
              const AMRAP = screen.getByTestId('FOR TIME');
              fireEvent.press(AMRAP);
              expect(navigation.navigate).toHaveBeenCalled();
            });
          });

          describe('And TABATA button is clicked', () => {
            test('Then navigation.navigate should have been called with handleClick', () => {
              const AMRAP = screen.getByTestId('TABATA');
              fireEvent.press(AMRAP);
              expect(navigation.navigate).toHaveBeenCalled();
            });
          });
          describe('And CRONO button is clicked', () => {
            test('Then navigation.navigate should have been called with handleClick', () => {
              const AMRAP = screen.getByTestId('CRONO');
              fireEvent.press(AMRAP);
              expect(navigation.navigate).toHaveBeenCalled();
            });
          });
        })
    })