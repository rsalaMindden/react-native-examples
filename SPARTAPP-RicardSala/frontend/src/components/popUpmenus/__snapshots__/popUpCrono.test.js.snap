// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Training component When it is render Then should match the snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "backgroundColor": "#595260",
        "flex": 1,
      },
      Object {
        "alignItems": "center",
      },
    ]
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#595260",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Object {
          "elevation": 15,
          "height": 155,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 7,
            "width": 0,
          },
          "shadowOpacity": 0.43,
          "shadowRadius": 9.51,
          "width": 349,
        }
      }
    >
      <Image
        resizeMode="cover"
        source={
          Object {
            "testUri": "../../../src/images/cronoImg.png",
          }
        }
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": 155,
              "width": 349,
            },
            Object {
              "borderRadius": 15,
            },
          ]
        }
      />
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#2C2E43",
          "borderRadius": 18,
          "fontSize": 24,
          "height": 35,
          "opacity": 1,
          "textAlign": "center",
          "width": 349,
        }
      }
      testID="FOR TIME"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#FFD523",
            },
            Object {
              "backgroundColor": "#2C2E43",
              "borderRadius": 18,
              "fontSize": 24,
              "height": 35,
              "textAlign": "center",
              "width": 349,
            },
          ]
        }
      >
        FOR TIME
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="TABATA"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#FFD523",
            },
            Object {
              "backgroundColor": "#2C2E43",
              "borderRadius": 18,
              "fontSize": 24,
              "height": 35,
              "textAlign": "center",
              "width": 349,
            },
          ]
        }
      >
        TABATA
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="CRONO"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#FFD523",
            },
            Object {
              "backgroundColor": "#2C2E43",
              "borderRadius": 18,
              "fontSize": 24,
              "height": 35,
              "textAlign": "center",
              "width": 349,
            },
          ]
        }
      >
        CRONO
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
